### DISCUSSION TITLE
How do I sync/download an Office 365 Sharepoint folder with one-time codes enabled

### DISCUSSION THREAD

**Discussion by bo-qeye** [abraunegg/onedrive#2010] [Q&A] [answered]
Hello

A client of ours is providing rather large datasets via an Office 365 Sharepoint site, and manual download via a browser fails - the resulting zip files are incomplete and corrupted.

I'm trying to connect to the site in question with this software, but so far I'm not having any luck.

The official procedure is that our client, who owns the Sharepoint site, has given access to johndoe@mycompany.com. We enter this email on a web page, and a one-time code is sent to that email address, which we can then use to access. As far as I can see, johndoe@mycompany.com does not have a OneDrive account, nor does he need one.

How do we make this work with abraunegg/onedrive?

Thanks in advance!

**Comment by abraunegg** [comment, marked as answer]
@bo-qeye 
> How do we make this work with abraunegg/onedrive?

You must authorise the client & authenticate the client as per https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#authorize-the-application-with-your-onedrive-account

This is the only way. Once this is done, configure the client to access the SharePoint site as per https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md - again, this is the only way.

Once done, the client will be able to download the data it has access to.

**Comment by bo-qeye** [comment]
Thank you for your prompt answer, unfortunately I don't have a user to authenticate, so that's a bit of a dead end...

I have successfully extracted the cookies from a browser session, and can download individual files with curl.

Is there a way to feed onedrive the cookies file and have it use that as authentication?

**Reply by abraunegg** [reply]
@bo-qeye 

No there is not. The only way to use this client is to authenticate the client in the prescribed manner.
